src/components/singlebar.js

```diff
@@ -1,41 +1,271 @@
+  // clean electrification. We do this by finding the non-electrifiable percent
+  // and then doing an inverse
+  const nonElectrificationPrcnt = getPct(emissionsData.dumps_farms_industrial_other, emissionsTotal)
+  const electrificationPrcnt = 100 * (1 - (nonElectrificationPrcnt / 100))
+
+  const activeFill = '#ff5722'
+
+  /**
+   * Configuration for the colors for each bar graph bar as well as their data
+   * key and label text
+   */
+  const BarsConfig = {
+    other: {
+      key: 'dumps_farms_industrial_other',
+      text: 'Farms, Industrial & Other',
+      fill: '#98886c',
+      // This categoy cannot be electrified so make the greenFill red to be
+      // clearly bad
+      greenFill: '#cabd98',
+    },
+    transport: {
+      key: 'transportation',
+      text: 'Transportation',
+      fill: '#a6a6a6',
+      greenFill: '#4caf50',
+    },
+    buildings: {
+      key: 'buildings',
+      text: 'Buildings',
+      fill: '#c2c2c2',
+      greenFill: '#6ebf70',
+    },
+    power: {
+      key: 'dirty_power',
+      text: 'Dirty Power',
+      fill: '#dcdcdc',
+      greenFill: '#a3d7a4',
+    }
+  }
+
+  const LabelOffset = 12
+  let LabelPosition = 'right'
+  let GraphMargin = {
+    top: 10,
+    right: 200,
+    left: 0,
+    bottom: 0
+  }
+
+  BarWidth = 150
+  GraphWidth = 400
+  GraphHeight = 350
+
+  // Apply custom styling with interior labels for homepage chart
+  if (homeView) {
+    LabelPosition = 'insideTopLeft'
+    BarWidth = 275
+    GraphWidth = 375
+    GraphMargin = {
+      top: 0,
+      right: 100,
+      left: 0,
+      bottom: 0
+    }
+  }
+
+  // If on state details, shorten text labels
+  if (!homeView) {
+    BarsConfig.power.text = 'Power'
+    BarsConfig.transport.text = 'Transport'
+    BarsConfig.other.text = 'Other'
+  }
+
+  // If greenKeys are specified, switch those bars' fill to their greenFill color
+  if (greenKeys) {
+    Object.values(BarsConfig).forEach(config => {
+      if (greenKeys.includes(config.key)) {
+        config.fill = config.greenFill
+      }
+    })
+  }
+
+  // If activeKey is specified, switch that bar's fill to its activeKey color
+  if (activeKey) {
+    const activeConfig = Object.values(BarsConfig)
+      .find(config => config.key === activeKey)
+
+    if (activeConfig) {
+      activeConfig.fill = activeFill
+    }
+    else {
+      console.error('Bad activeKey!', activeKey)
+    }
+  }
+
+  return (
+    <div className="single-bar-chart">
+      <BarChart
+        barSize={ BarWidth }
+        width={ GraphWidth }
+        height={ GraphHeight}
+        data={[emissionsData]}
+        margin={ GraphMargin }
+      >
+        { /* Make sure the y-axis matches the data exactly so the bars take up 100% of the height */ }
+        <YAxis domain={['dataMin', 'dataMax']} hide={ true } />
+        {
+          // Only show other bar if it's non-zero
+          emissionsData[BarsConfig.other.key] &&
```

Comment:
in future PRs we should make this more DRY as the only thing that's really changing is the key we're referencing